### PR TITLE
Starter content: parallelise post creation; tweaks for handling E2E

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR introduces an optimisation of starter content creation by parallelising requests. 

This PR also introduces a couple of changes to enable E2E testing:
- handling `id` attribute on `StyleCard` and using it on theme selection cards
- using URL query param instead of option to achieve deterministic content for E2E
- using the same image for all E2E posts
- hardcoded post publication date for E2E posts

### How to test the changes in this Pull Request:

1. Visit `/wp-admin/admin.php?page=newspack-setup-wizard#/starter-content` and create some starter content. Is should be faster than than before, but other than that with no changes.
2. Set `WP_NEWSPACK_DETERMINISTIC_STARTER_CONTENT` env var to `1` (e.g. `define( 'WP_NEWSPACK_DETERMINISTIC_STARTER_CONTENT', '1' );` in `wp-config.php`)
3. Run the starter content again. It should create just 12 posts, all with the same image, title, and content.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-starter-content.php [hunk 1]
```diff
@@ -359,7 +359,8 @@ private static function add_featured_image( $post_id, $post_index ) {
 			require_once ABSPATH . 'wp-admin/includes/image.php';
 		}
 
-		$url = self::is_e2e() ? 'https://picsum.photos/id/' . $post_index . '/1200/800' : 'https://picsum.photos/1200/800';
+		// Use same image everywhere for e2e.
+		$url = self::is_e2e() ? 'https://picsum.photos/id/424/1200/800' : 'https://picsum.photos/1200/800';
```
Comment (claudiulodro) [inline includes/class-starter-content.php]: This is a very nice picture!

Comment (adekbadek) [reply includes/class-starter-content.php]: My first shot was [id 42](https://picsum.photos/id/42/1200/800), but it was not the answer to life, universe, and everything. This is is closer to it IMO.

**Comment by adekbadek** [comment]
@claudiulodro I've added two things:
1. categories derived from post index (there'll be 12 posts for e2e dummy content, one per category): 
ad263ae
1. disabling background plugin installation in setup for e2e: 8237503

**Comment by jeffersonrabb** [comment]
I ran a test with the E2E flag on. When viewing the edit page for each post, the Featured Image is correct (https://picsum.photos/id/424/1200/800). However, when previewing the homepage immediately after, the images are all different (although dates, titles, and excerpts are the same). Any idea why this would be? Before running the Starter Content Wizard I deleted all posts, media, and categories.

_Note: Ignore the datestamp of the top featured story in the screenshot — I accidentally updated. All the others still show a variety of images which do not depict a cabin in the woods._

Screen Shot 2020-03-22 at 9 42 15 PM

**Comment by adekbadek** [comment]
@jeffersonrabb  maybe that's Jetpack images caching? Assuming `newspack1.ngrok.io` was used before for other stuff.

**Comment by jeffersonrabb** [comment]
> @jeffersonrabb maybe that's Jetpack images caching? Assuming `newspack1.ngrok.io` was used before for other stuff.

Yup, I turned off JP image accelerator and the images are all the same.

File: assets/wizards/setup/index.js [hunk 2]
```diff
@@ -272,6 +276,15 @@ class SetupWizard extends Component {
 			'/theme-style-selection',
 			'/starter-content',
 		];
+
+		// background auto installation is a nice feature, but in e2e it
+		// cretes an undeterministic environment, since the installation-progress
```
Comment (jeffersonrabb) [inline assets/wizards/setup/index.js]:
```suggestion
		// creates an undeterministic environment, since the installation-progress
```

**Pull request merged**